src/nix/profile.cc

```diff
@@ -149,6 +154,25 @@ struct ProfileManifest
                         e["outputs"].get<ExtendedOutputsSpec>()
                     };
                 }
+
+                std::string nameCandidate = element.identifier();
+                if (e.contains("name")) {
+                    nameCandidate = e["name"];
+                }
+                else if (element.source) {
+                    auto url = parseURL(element.source->to_string());
+                    auto name = getNameFromURL(url);
+                    if (name)
+                        nameCandidate = *name;
+                }
+
+                auto finalName = nameCandidate;
+                for (int i = 1; foundNames.contains(finalName); ++i) {
+                    finalName = nameCandidate + std::to_string(i);
```

Comment:
My initial fear was that this change might cause issues if entry names match, but clearly this was considered :)

I must have misread this part of your PR message as well, which says as much:

> Duplicate packages are handled by suffixing the name with a number. This is simple and effective, and I assume we don't hit that case too often in reality.

Reply:
To be fair, the PR message is *very* long, so I can see that details like this might be skipped over when reading it.